lnbits/core/views/api.py

```diff
@@ -99,8 +99,7 @@ async def api_update_balance(
 
 @core_app.put("/api/v1/wallet/{new_name}")
 async def api_update_wallet(
-    new_name: str, wallet: WalletTypeInfo = Depends(WalletAdminKeyChecker())
-):
+    new_name: str, wallet: WalletTypeInfo = Depends(get_key_type)):
```

Comment:
If you use `get_key_type` you must check manually if the wallet is valid. The `wallet.id` will be 2 if it is invalid. `update_wallet` will not check this either.

Reply:
I have just edited, as it worked few months before. And it works now.

Reply:
looks like this was changed in #403, I'm fairly sure you would want to use `WalletAdminKeyChecker()`

Reply:
That way it won't work, as it is in current version, so I changed it back

Reply:
do you mean that the [main branch](https://github.com/lnbits/lnbits-legend/blob/main/lnbits/core/views/api.py#L102) broken currently? If there is something in the extension that is not working with this, that should be corrected because this file is outside of the extension and a core lnbits file

Reply:
Yes the main branch is broken, this change doesn't has something to do with the extension, I just wanted to fix that in the core code but didn't knew it will pop up in that PR.

Reply:
What is the actual error you're seeing? WalletAdminKeyChecker checks if a valid admin key was supplied. You shouldn't be able to rename a wallet with just an invoice key.

Reply:
It's not working on latest version of LNBits, just check it without my changes.

That's the error and the issue: https://github.com/lnbits/lnbits-legend/issues/514

I knew it was working before so I looked back in time.

Reply:
OK, thanks. I'll take a look.

Reply:
Just removed the fix of renaming wallet, theres a proper fix in an other PR https://github.com/lnbits/lnbits-legend/pull/588.